File: library/src/main/java/me/proxer/library/enums/ChatMessageAction.java
package me.proxer.library.enums;

import com.serjltt.moshi.adapters.FallbackEnum;
import com.squareup.moshi.Json;
import me.proxer.library.entity.chat.ChatMessage;

/**
 * Enum holding the possible actions of a {@link ChatMessage}.
 *
 * @author Ruben Gees
 */
@FallbackEnum(name = "NONE")
public enum ChatMessageAction {
    @Json(name = "") NONE,
    @Json(name = "removeMessage") REMOVE_MESSAGE,
}

File: library/src/main/kotlin/me/proxer/library/enums/ChatMessageAction.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import me.proxer.library.entity.chat.ChatMessage

/**
 * Enum holding the possible actions of a [ChatMessage].
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = false)
@FallbackEnum(name = "NONE")
enum class ChatMessageAction {

    @Json(name = "")
    NONE,

    @Json(name = "removeMessage")
    REMOVE_MESSAGE
}
